inst/tutorials/r_and_rstudio_basic/r_and_rstudio_basic.Rmd

```diff
@@ -0,0 +1,284 @@
+
+RStudio has a simple interface to create and switch between projects, accessed from the button in the top-right corner of the RStudio window. (Labeled "Project: (None)", initially.)
+
+Let's create a project to work in for this tutorial. Start by clicking the "Project" button in the upper right or going to the "File" menu. Select "New Project", and the following will appear:
+
+![](/images/create_project.png){width=75%}
+
+
+Choose "New Directory" followed by "New Project" and click on "Browse...". Navigate to your Desktop, and name the directory `<course_name> R`(replace `<course_name>` with the name of your class, e.g. `MICB301`) for this project.
+
+After your project is created, navigate to its directory using your Finder/File explorer or the integrated Terminal in RStudio. You will see the ".RProj" file has been created.
+
+You can open this project in the future in one of three ways:
+
+- In your file browser (e.g. Finder or Explorer), simply double-click on the `.RProj` file 
+- In an open RStudio window, choose "File" &rarr; "Open Project"
+- Switch among projects by clicking on the R project symbol in the upper left
+corner of RStudio
+
+
+
+## Variables in R
+
+We use variables to store data that we want to access or manipulate later. Variables must have unique names.  
+
+Without declaring a variable the sum of these two numbers will be printed to console but cannot be accessed for future use:
+
+```{r}
+2 + 2 
+```
+
+To declare a variable, follow the pattern of: `variable <- value`. Let's declare a variable `total` as the sum of two numbers.
+
+```{r}
+total <- 2 + 2
+```
+
+We access the value of `total`:
+
+```{r}
+total
+```
+
+We can use the value stored in `total`:
+
+```{r}
+total - 1
+```
+
+After declaring a variable, we can perform operations to change the value stored in the variable:
+
+```{r}
+total <- total - 1
+```
+
+Now `total` stores a new value:
+
+```{r}
+total
+```
+
+
+
+## Functions
+
+Functions are one of the basic units in programming. Generally speaking, a function takes some input and generates some output. Every R function follows the same basic syntax, where `function()` is the name of the function and `arguments` are the different parameters you can specify (i.e. your input):
+
+`function(argument1 = ..., argument2 = ..., ...)`
+
+You can treat functions as a black box and do not necessarily need to know how it works under the hood as long as your provided input conforms to a specific shape.
+
+![](/images/function.png){width=75%}
+
+For example, the function `sum()` expects numbers:
+
+```{r}
+sum(3, 5, 9, 18)
+```
+
+If you instead pass text as arguments to `sum()` you will receive an error:
+
+```{r error = TRUE}
+sum("Sum", "does", "not", "accept", "text!")
+```
+
```

Comment:
```suggestion
The use of functions isn't limited to mathematical calculations. Function can also be used to transform data.

For example, the function `t()` can be used to transpose a matrix. In the example below, we generate a matrix (`example_matrix`) using the functions `c()` and `as.matrix()`, which were used to combine values into a vector and transform the vector into a matrix. We then use the `t()` function to transpose `example_matrix` into the matrix `example_transposed`.

```{r t_function, exercise=TRUE}
# Generate Matrix
example_list <- c("T", "does", "accept", "text!")
example_matrix <- as.matrix(example_list)

#Transpose Matrix
example_transposed <- t(example_matrix)

#Display Original and Transposed Matrices
example_matrix
example_transposed
```
### The most helpful function of all: the 'help' function
You can get information about a specific function by running the command `?<function>` or `help(<function>)` (replace `<function>` by the name of the function you are interested in). This command opens the help page, where you can find all information about a function's purpose and its arguments. For beginners, it is useful to concentrate on the "Examples" and "Arguments" section to understand the typical usage of the function better.

Run the code below to read the documentation for the `t()` function.

```{r transpose_help, exercise=TRUE}
?t
```
```

Reply:
I copied this code over from my .Rmd file, but it seems like the "suggested change" syntax interferes with the RMarkdown syntax.  Will this be a problem when we accept the changes?